pkg/prometheus/server/operator.go

```diff
@@ -1161,9 +1162,19 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 	}
 
 	p.Status = *pStatus
+	selectorLabels := makeServerSelectorLabels(p.Name)
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: selectorLabels})
+	if err != nil {
+		return fmt.Errorf("failed to create selector for prometheus scale status: %w", err)
+	}
+	p.Status.Selector = selector.String()
+	p.Status.Shards = ptr.Deref(p.Spec.Shards, 1)
```

Comment:
nit-picky but we factorize a bit more and have a single function in pkg/prometheus doing it (as long as it gets the selector labels as input).